src/test/java/org/jenkinsci/plugins/workflow/libs/SCMSourceRetrieverTest.java

```diff
@@ -124,6 +129,25 @@
             WorkflowRun b = r.buildAndAssertSuccess(p);
             List<ChangeLogSet<? extends ChangeLogSet.Entry>> changeSets = b.getChangeSets();
             assertEquals(0, changeSets.size());
+            r.assertLogNotContains("Retrying after 10 seconds", b);
         }
     }
+
+    @Test public void retry() throws Exception {
```

Comment:
It would be nice to have a test with a FailingSCMSource that started working after the first failure, so you could use `r.jenkins.setScmCheckoutRetryCount(2);` and check that the build completes succefully.